sys/src/9/amd64/multiboot.c

```diff
@@ -164,17 +181,29 @@ multiboot(u32 magic, u32 pmbi, int vflag)
 			if(mod->string != 0)
 				p = KADDR(mod->string);
 			if(vflag)
-				print("mod %#x %#x <%s>\n",
+				print("Multiboot: mod %#x %#x <%s>\n",
 				      mod->modstart, mod->modend, p);
 			else {
+				u64 modstart = mod->modstart;
 				usize len = mod->modend - mod->modstart;
-				pamapinsert(mod->modstart, len, PamMODULE);
+				pamapinsert(modstart, ROUNDUP(len, PGSZ), PamMODULE);
+
+				if (!initrd) {
```

Comment:
what if initrd is set? Isn't that some kind of error? What does this test mean? I think a comment would be helpful.